src/Data/Vec/Functional/Algebra/Base.agda

```diff
@@ -0,0 +1,46 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Some Vector-related module Definitions
+------------------------------------------------------------------------
+
+{-# OPTIONS --cubical-compatible --safe #-}
+
+open import Function using (_$_)
+open import Data.Product hiding (map)
+open import Data.Nat using (ℕ)
+open import Data.Fin using (Fin)
+open import Data.Vec.Functional
+open import Algebra.Core
+open import Algebra.Bundles
+open import Algebra.Module
+open import Relation.Binary
+import Data.Vec.Functional.Relation.Binary.Equality.Setoid as VecSetoid
+import Algebra.Definitions as AD
+import Algebra.Structures as AS
+
+module Data.Vec.Functional.Algebra.Base
+  {c ℓ} (ring : Ring c ℓ) where
+
+private variable
+  n : ℕ
+
+open Ring ring
+open VecSetoid setoid
+
+VC = Vector Carrier
```

Comment:
I don't think we should have this definition. It adds another layer of indirection without adding any clarity.

Reply:
removed it